R/fdbk_asdataframe.R

```diff
@@ -1203,6 +1261,133 @@ fdbk_dt_binning <- function(DT,varToBin="level",binLower,binUpper){
+#' a2 = -9 # K/km lapse rate obtained from a fictive model output
+#' b1 = 288.15 # K standard atmosphere surface temperature
+#' b2 = 295 # K surface temperature obtained from a fictive model output
+#' Ho = 8.4 # km scale height
+#' po = 1013.25 # standard atmosphere pressure in hPa
+#' p = seq(250,1000,10) # pressure until the tropopause
+#' T1 = a1*Ho*log(po/p)+b1 # Standard amtmosphere temperature profile
+#' T2 =  a2*Ho*log(po/p)+b2 # Model output temperature profile 
+#' Bias = T2-T1 # Bias = forecast - observation 
+#' 
+#' # Build a data table in feedback files format
+#' obs = T1
+#' veri_data = T2
+#' veri_forecast_time = 24
+#' veri_initial_date = 2015110900
+#' time = -720
+#' lat = 46.812
+#' lon = 6.943
+#' varno = 2
+#' veri_model = "COSMO"
+#' plevel = p
+#' ident = 6610
+#' levels = c(1000, 975, 950, 925, 900, 875, 850, 800, 750, 700, 650, 600, 550, 500, 450, 400, 350, 300, 250) # Levels on which to interpolate
+#' DT = data.frame(obs,veri_data,veri_forecast_time,veri_initial_date,time,lat,lon,varno,veri_model,plevel,ident)
+#' DT           = fdbk_dt_interpolate(DT,varToInter=c("obs","veri_data"), levelToInter = "plevel", interLevels = levels) # interpolate DT
+#'
+#' data1 = melt(data.frame(T1,p),id="T1") # Data for the standard atmosphere temperature profile
+#' data2 = melt(data.frame(T2=DT$obs,DT$plevel),id="T2") # Interpolation of data1 
+#' 
+#' 
+#' plot = ggplot() + geom_point(data=data1,aes(x=T1,y=value,colour=variable)) + geom_point(data=data2,aes(x=T2,y=value,colour=variable))+scale_y_reverse()+
+#'   xlab("T [K]") + ylab("pressure [hPa]")+  scale_colour_manual(name="Temperature",values=c("red","blue"),labels=c("Interpolation","Standard atmosphere"))+theme(legend.position = "top")
+#' print(plot) # plot of the Standard atmosphere profile and its interpolation 
+#' 
+#' allscores = fdbk_dt_verif_continuous(DT,strat=c("varno","veri_model","plevel") ) # Data table with scores
+#' 
+#' data3 = melt(data.frame(Bias,p),id="Bias") # Bias calculated directly from the standard atmosphere and model output profiles
+#' ME = allscores[allscores$scorename=="ME"]$scores # scores calculated with fdbk_dt_verif_continuous on interpolation levels
+#' ME_levels = allscores[allscores$scorename=="ME"]$plevel # interpolation levels
+#' data4 = melt(data.frame(ME,ME_levels),id="ME")
+#' plot2 = ggplot() + geom_point(data=data3,aes(x=Bias,y=value,colour=variable)) + geom_point(data=data4,aes(x=ME,y=value,colour=variable))+scale_y_reverse()+
+#'   xlab("T bias [K]") + ylab("pressure [hPa]")+scale_colour_manual(name="Bias",values=c("red","blue"),labels=c("Interpolation","Standard atmosphere"))+theme(legend.position = "top")
+#' print(plot2) # plot of the bias calculated directly from the profiles and the bias from the interpolation
+
+fdbk_dt_interpolate <- function(DT,varToInter=c("obs","veri_data"), levelToInter = "plevel", interLevels = levels, varno="varno"){
+  
+  #OneSounding = data.frame(,nrow=1,ncol=dim(DT)[2])
+  ncol = dim(DT)[2]
```

Comment:
I did not know about the function ncol. Thanks for the tip, I changed the name to Ncol